flintrock/flintrock.py

```diff
@@ -316,6 +316,7 @@ def cli(cli_context, config, provider, debug):
               multiple=True,
               help="Additional tags (e.g. 'Key,Value') to assign to the instances. "
                    "You can specify this option multiple times.")
+@click.option('--client-source', type=str, default=None)
```

Comment:
I think we need to improve a few things here.

1. I'd rename this option to `--ec2-authorize-access-from`. It's clearer, and it categorizes the option under EC2 where it belongs.
2. I'd also add a brief help text explaining what this option does and what input it takes, since it may not be obvious to everyone. Since this option disables the automatic detection of the client IP, we should call that out too.
3. We should somehow validate the input here since we are expecting either a CIDR or a security group ID. A user might, as I did, assume they can provide a plain IP address, which then leads to the following error:

    ```
    botocore.exceptions.ClientError: An error occurred (InvalidParameterValue) when calling the 
    AuthorizeSecurityGroupIngress operation: CIDR block 172.31.38.197 is malformed
    ```

    Either that, or we should accept a plain IP address and automatically covert it into a `/32` CIDR block.

    Take a look at `--ec2-tag`  for an example of how to validate user input.
4. Since we are adding a new option, we need to add it to the [config template](https://github.com/nchammas/flintrock/blob/master/flintrock/config.yaml.template).